tests/pkg_openwsn/README.md

```diff
@@ -0,0 +1,311 @@
+    - reduce SLOTFRAME_LENGTH
+
+If you want to disable channel hopping to speed up synchronization you can set
+a fix channel.
+
+    $ export CFLAGS=-DIEEE802154E_SINGLE_CHANNEL=17
+
+Enhanced beacon are sent once per slotframe with a likelihood of 1/EB_PORTION.
+The default value is EB_PORTION = 10 (10%), reducing the value of EB_PORTION
+increases the frequency at which beacons are sent, this will also help nodes
+to stay synchronized.
+
+    $ export CFLAGS=-DEB_PORTION=4
+
+EB gets an opportunity to be sent (depending on 1/EB_PORTION) once in every
+slotframe, reducing the soltframe length will cause EB to be sent more often.
+Beware that this could have an impact on the MSF (Minimal Scheduling Function).
+
+    $ export CFLAGS=-DSLOTFRAME_LENGTH=51
+
+See [documentation](../../pkg/openwsn/doc.txt#Synchronization) for more on
+synchronization.
+
+## IMPORTANT!
+
+OpenWSN uses source routing and this means all network traffic must go through
+from the root node to OpenVisualizer. If the root node configuration can't
+handle the configured baudrate correctly this will lead to packet loss.
+
+Currently these are the tested configurations:
+
+(a) samr21-xpro network:
+    - leaf nodes using `openwsn_sctimer_rtt`
+    - root node using `openwsn_sctimer_rtt` and 19200 baudrate directly
+      connected to UART pins (not through the usb debugger)
+
+(b) iotlab-m3 network:
+    - leaf nodes using `openwsn_sctimer_rtt` or `sctimer_ztimer`
+    - root node using `openwsn_sctimer_rtt` and 57600 baudrate
+
+For more details on this please refer to [pkg documentation](../../pkg/openwsn/doc.txt).
+
+## Testing configuration (a) (three local samr21-xpro nodes)
+
+1. figure out each node's serial number (`edbg -l` output might help)
+2. hook up the root node to an external USB-serial-converter. For that, connect
+   the converter's RXD/TXD pins to the node's PA22/PA23.
+3. flash the root node:
+
+        $ SERIAL=${ROOT_SERIAL_NODE} OPENSERIAL_BAUD=19200 USEMODULE=openwsn_serial \
+              BOARD=samr21-xpro make flash -j4
+
+4. flash the leaf nodes:
+
+        $ BOARD=samr21-xpro make all -j4
+        $ BOARD=samr21-xpro SERIAL=${LEAF_SERIAL_NODE0} make flash-only
+        $ BOARD=samr21-xpro SERIAL=${LEAF_SERIAL_NODE1} make flash-only
+
+5. open a shell to the leaf nodes
+   so in two shell windows, do (one in each):
+
+        $ BOARD=samr21-xpro SERIAL=${LEAF_SERIAL_NODE0} make term
+        $ BOARD=samr21-xpro SERIAL=${LEAF_SERIAL_NODE1} make term
+
+6. in a third shell, launch openvisualizer:
+
+        $ BOARD=samr21-xpro PORT=<USB-serial-port, e.g., /dev/ttyUSB0> BAUD=19200 \
+              make openv-termroot
+
+## Testing configuration (b) (iotlab-m3 network on iotlab)
+
+When using OpenVisualizer over iot-lab an ssh-tunnel is opened to connect to the
+IoT-LAB motes' TCP port. For this to work you will need:
+
+- A valid IoT-LAB [account](https://www.iot-lab.info/testbed/signup)
+- Authenticate to IoT-LAB `$ iotlab-auth -u <login>`
```

Comment:
Should this be done locally or on a SSH frontend ?

Reply:
Locally